classes/image_type_tezi_run.bbclass

```diff
@@ -1,26 +1,31 @@
-inherit image_types
+python () {
+    if bb.utils.contains("IMAGE_FSTYPES", "tezirunimg", "1", "0", d) == "1":
+        bb.build.addtask('do_assemble_tezirunimg', 'do_image_complete', 'do_image_squashfs', d)
+}
```

Comment:
Yes, usually it could be handled by IMAGE_TYPEDEP, but the tricky part here is, the do_image_tezirunimg task is a bundle of python tasks and shell tasks, but IMAGE_CMD only support shell tasks, so we can not make do_image_tezirunimg a IMAGE_CMD, and we also can not make it to be a empty IMAGE_CMD as well, since there are quite some processes in image.bbclass to add prefuncs and so on to IMAGE_CMD, they could not work with RM_WORK, for instance, the set_image_size task requires IMAGE_ROOTFS, but it does not always exist since RM_WORK is being enabled, which means the IMAGE_CMD might fail if its task hash changed(or its dependent tasks' hashes) while the do_rootfs task hash not changed. So we have to mask the processes in image.bbclass with IMAGE_TYPES_MASKED += "tezirunimg", so that's why we need add the task dependency do_assemble_tezirunimg -> do_image_squashfs in this way.